File: ios/StockPro/Data/DTOs/SupplierDTOs.swift
import Foundation

struct SupplierDTO: Decodable, Identifiable {
    let id: Int
    let name: String
    let email: String?
    let phone: String?
    let address: String?
    let contact_person: String?

    func toItem() -> SupplierListItem {
        SupplierListItem(
            id: id,
            name: name,
            email: email ?? "",
            phone: phone ?? "",
            address: address ?? "",
            contactPerson: contact_person ?? ""
        )
    }
}

struct SupplierCreateRequest: Encodable {
    let name: String
    let email: String?
    let phone: String?
    let address: String?
    let contact_person: String?
}

struct SupplierUpdateRequest: Encodable {
    let name: String
    let email: String?
    let phone: String?
    let address: String?
    let contact_person: String?
}

struct SupplierListItem: Identifiable {
    let id: Int
    let name: String
    let email: String
    let phone: String
    let address: String
    let contactPerson: String
}

File: android/core/data/src/main/java/com/app2/core/data/remote/dto/SupplierDTOs.kt
package com.app2.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SupplierDTO(
    val id: Int,
    val name: String,
    val email: String? = null,
    val phone: String? = null,
    val address: String? = null,
    @SerialName("contact_person") val contactPerson: String? = null,
    @SerialName("created_at") val createdAt: String? = null
)

@Serializable
data class SupplierCreateRequest(
    val name: String,
    val email: String? = null,
    val phone: String? = null,
    val address: String? = null,
    @SerialName("contact_person") val contactPerson: String? = null
)

@Serializable
data class SupplierUpdateRequest(
    val name: String? = null,
    val email: String? = null,
    val phone: String? = null,
    val address: String? = null,
    @SerialName("contact_person") val contactPerson: String? = null
)
